Microsoft.Toolkit.Uwp.UI.Controls/WrapPanel/WrapPanel.Data.cs

```diff
@@ -33,6 +41,55 @@ public UvMeasure(Orientation orientation, double width, double height)
                     V = width;
                 }
             }
+
+            public UvMeasure Add(double u, double v)
+                => new UvMeasure { U = U + u, V = V + v };
+        }
+
+        private struct UvRect
+        {
+            public UvMeasure Position { get; set; }
+
+            public UvMeasure Size { get; set; }
+
+            public Rect ToRect(Orientation orientation)
+            {
+                switch (orientation)
+                {
+                    case Orientation.Vertical:
+                        return new Rect(Position.V, Position.U, Size.V, Size.U);
+                    case Orientation.Horizontal:
+                        return new Rect(Position.U, Position.V, Size.U, Size.V);
+                    default:
+                        throw new NotSupportedException();
```

Comment:
This should be using the `ThrowHelper` APIs to get better codegen, or that `throw new` will bloat the asm a bit.

```suggestion
                    default: return ThrowHelper.ThrowNotSupportedException<Rect>();
```